controllers/workspace/metrics/failure_reason.go

```diff
@@ -15,15 +15,13 @@
 
 package metrics
 
-type FailureReason struct {
-	reason string
-}
+type FailureReason string
 
-var (
-	ReasonBadRequest             = FailureReason{reason: "BadRequest"}
-	ReasonInfrastructureFailure  = FailureReason{"InfrastructureFailure"}
-	ReasonWorkspaceEngineFailure = FailureReason{"WorkspaceEngineFailure"}
-	ReasonUnknown                = FailureReason{"Unknown"}
+const (
+	ReasonBadRequest             = "BadRequest"
+	ReasonInfrastructureFailure  = "InfrastructureFailure"
+	ReasonWorkspaceEngineFailure = "WorkspaceEngineFailure"
+	ReasonUnknown                = "Unknown"
```

Comment:
This keeps the constants as string-type (and requires the workarounds below). It'd be simpler to just use
```suggestion
const (
	ReasonBadRequest             FailureReason = "BadRequest"
	ReasonInfrastructureFailure  FailureReason = "InfrastructureFailure"
	ReasonWorkspaceEngineFailure FailureReason = "WorkspaceEngineFailure"
	ReasonUnknown                FailureReason = "Unknown"
)
```

The purpose of using a type here is to lightly restrict the different failure reasons -- if a function expects a defined FailureReason as a parameter, it prompts you to use one of the defined constants. When eventually the failureReason as-a-string is required, we can do `string(reason)`.

Reply:
Oops, thanks @amisevsk . I made the changes and did some small refactoring.